aten/src/ATen/gen.py

```diff
@@ -368,10 +368,11 @@ def generate_storage_type_and_tensor(backend, density, declarations, per_op_regi
     if env['DeviceType'] == 'CPU':
         top_env['cpu_type_headers'].append(
             '#include <ATen/{}.h>'.format(env['Type']))
-    else:
-        assert env['DeviceType'] == 'CUDA'
+    elif env['DeviceType'] == 'CUDA':
         top_env['cuda_type_headers'].append(
             '#include <ATen/{}.h>'.format(env['Type']))
+    else:  # Allow gen.py to be called by out-of-tree devices
+        pass
```

Comment:
Oof, this is gonna be rough to support in the long term. @dylanbespalko, did you try doing this manually without gen.py? What were the barriers here?

Reply:
I knew this would be scary :). This allows me to run gen.py from my out-of-tree extension so that I can update myself whenever `native_functions.yaml`. When building pytorch, the solution works for me, but I will need to verify this with CI.